src/main/java/org/folio/rest/impl/UserTenantsAPI.java

```diff
@@ -35,12 +39,42 @@ public void getUserTenants(String userId, String username, String tenantId, int
     String okapiTenantId = TenantTool.tenantId(okapiHeaders);
     Criterion criterion = new Criterion().setLimit(new Limit(limit)).setOffset(new Offset(offset));
     addWhereClauseArgumentsToCriterion(userId, username, tenantId, criterion);
+    logger.debug("Trying to get user-tenant records with criterion: {}.", criterion);
 
     userTenantService.fetchUserTenants(okapiTenantId, criterion, vertxContext.owner())
-      .onSuccess(res -> asyncResultHandler.handle(
-        succeededFuture(GetUserTenantsResponse.respond200WithApplicationJson(res))))
-      .onFailure(cause -> asyncResultHandler.handle(
-        succeededFuture(GetUserTenantsResponse.respond500WithTextPlain(cause.getMessage()))));
+      .onSuccess(res -> {
+        logger.info("Number of existing user-tenant records: {}.", res.getTotalRecords());
+        asyncResultHandler.handle(
+          succeededFuture(GetUserTenantsResponse.respond200WithApplicationJson(res)));
+      })
+      .onFailure(cause -> {
+        String message = cause.getLocalizedMessage();
+        logger.error(message, cause);
+        asyncResultHandler.handle(
+          succeededFuture(GetUserTenantsResponse.respond500WithTextPlain(cause.getMessage())));
+      });
+  }
+
+  @Override
+  public void postUserTenants(String lang, UserTenant userTenant, Map<String, String> okapiHeaders,
+                              Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    String okapiTenantId = TenantTool.tenantId(okapiHeaders);
+    logger.debug("Trying to save user-tenant with id: {}, userId: {}, userName: {}, tenantId: {}.",
+      userTenant.getId(), userTenant.getUserId(), userTenant.getUserName(), userTenant.getTenantId());
+
+    userTenantService.saveUserTenant(userTenant, okapiTenantId, vertxContext.owner())
+      .onSuccess(res -> {
+        logger.info("user-tenant with id: {}, userId: {}, userName: {}, tenantId: {} has been saved successfully.",
+          userTenant.getId(), userTenant.getUserId(), userTenant.getUserName(), userTenant.getTenantId());
+        asyncResultHandler.handle(
+          succeededFuture(Response.status(201).build()));
+      })
+      .onFailure(cause -> {
+        String message = cause.getLocalizedMessage();
+        logger.error(message, cause);
```

Comment:
lets also use predefined message in both endpoints, because getLocalizedMessage we will see from stacktrace, but if we want search log by some predefined message - it would be easier

Reply:
Fixed